src/components/line-chart-audification/line-chart-audification.component.ts

```diff
@@ -1,25 +1,116 @@
-import { Component, Host } from '@angular/core';
-import { LineChartComponent } from '../line-chart/line-chart.component';
-import { AudificationService } from '../../services/audification/audification.service';
-import * as Tone from 'tone';
+import { Component, EventEmitter, HostListener, Input, NgZone, OnChanges, OnDestroy, Output, SimpleChanges } from '@angular/core';
+import { Melody } from '../../models/melody/melody.model';
+import { AUDIFICATION, t, tA11y } from '../../assets/i18n';
+import { Datum } from '../../d3/xy-chart.d3';
+import { formatX, formatY } from '../../utils/formatters';
 
 @Component({
   selector: 'app-line-chart-audification',
   templateUrl: './line-chart-audification.component.html',
+  styleUrls: ['./line-chart-audification.component.scss'],
 })
-export class LineChartAudificationComponent {
-  public disposeMelody?: () => void;
+export class LineChartAudificationComponent implements OnDestroy, OnChanges {
+  @Input() data: Datum[];
+  @Input() activeDatum: Datum | null;
+  @Output() activeDatumChange = new EventEmitter<Datum | null>();
+  @Input() frequencyRange: [number, number] = [256, 2048];
+  @Input() duration = 5;
+  liveText: string | null = null;
+  private melody?: Melody;
+  private domain: Date[];
+  private range: number[];
 
   constructor(
-    @Host() public component: LineChartComponent,
-    private audificationService: AudificationService,
+    private zone: NgZone,
   ) {
+    this.handleSeek = this.handleSeek.bind(this);
```

Comment:
You could instead use the [autobind decorator](https://github.com/jayphelps/core-decorators#autobind)